### PR TITLE
kodi-rbp: fix #1140 - no proper seat for kodi user

### LINKED ISSUE #1140: kodi-rbp: included systemd service does not allow kodi user to restart or shutdown
Package: kodi-rbp 14.1-4
Architecture: both armv6 and armv7
Device: RPi and RPi2

What it isn't doing: Interacting with systemd to allow Kodi to control system restart or shutdown.

What it should be doing: Allowing the system to restart or to shutdown from the Kodi GUI.

Steps to reproduce:
1) Install this package on RPi or RPi2
2) Start the kodi service
3) Try to restart or shutdown from the Kodi GUI (arrow down to the power button icon and hit ENTER)

### PR DESCRIPTION
**Pull request by graysky2**
Changes: fix to kodi.service and addition of required deps to PKGBUILD
Why: the current service file provides partial functionality to the software, ie what is described in https://github.com/archlinuxarm/PKGBUILDs/issues/1140

### CODE REVIEW

File: alarm/kodi-rbp/PKGBUILD [hunk 1]
```diff
@@ -8,12 +8,12 @@ buildarch=20
...
 pkgver=14.1
 _codename=Helix
-pkgrel=4
+pkgrel=5
 pkgdesc="A software media player and entertainment hub for digital media for the Raspberry Pi"
 arch=('armv6h' 'armv7h')
 url="http://kodi.tv"
 license=('GPL' 'custom')
-depends=('hicolor-icon-theme' 'fribidi' 'lzo2' 'smbclient' 'libtiff' 'libva' 'libpng' 'libcdio' 'yajl' 'libmariadbclient' 'libjpeg-turbo' 'libsamplerate' 'libssh' 'libmicrohttpd' 'sdl_image' 'python2' 'libass' 'libmpeg2' 'libmad' 'libmodplug' 'jasper' 'rtmpdump' 'unzip' 'xorg-xdpyinfo' 'libbluray' 'libnfs' 'afpfs-ng' 'avahi' 'bluez-libs' 'tinyxml' 'raspberrypi-firmware' 'libcec-rpi' 'libplist' 'swig' 'taglib' 'libxslt' 'shairplay-git')
+depends=('hicolor-icon-theme' 'fribidi' 'lzo2' 'smbclient' 'libtiff' 'libva' 'libpng' 'libcdio' 'yajl' 'libmariadbclient' 'libjpeg-turbo' 'libsamplerate' 'libssh' 'libmicrohttpd' 'sdl_image' 'python2' 'libass' 'libmpeg2' 'libmad' 'libmodplug' 'jasper' 'rtmpdump' 'unzip' 'xorg-xdpyinfo' 'libbluray' 'libnfs' 'afpfs-ng' 'avahi' 'bluez-libs' 'tinyxml' 'raspberrypi-firmware' 'libcec-rpi' 'libplist' 'swig' 'taglib' 'libxslt' 'shairplay-git' 'xorg-xinit' 'xorg-server' 'xf86-video-fbdev')
```
Comment (kmihelich) [inline alarm/kodi-rbp/PKGBUILD]: This package now explicitly depends on xorg-xinit, xorg-server, xf86-video-fbdev in order to run?

**Comment by kmihelich** [comment]
Why does this not apply to kodi-rbp-git?

File: alarm/kodi-rbp/PKGBUILD (hunk 1 above)
Comment (graysky2) [inline alarm/kodi-rbp/PKGBUILD]: Yes, using the modified service, these are indeed requirements on my RPi.

Comment (kmihelich) [inline alarm/kodi-rbp/PKGBUILD]: This package only works with the service now?

**Comment by graysky2** [comment]
Once we confirm that my PR indeed fixes this issue on a fresh install as I noted in https://github.com/archlinuxarm/PKGBUILDs/issues/1140, we can then apply it to the git package.

**Comment by kmihelich** [comment]
Why have you not confirmed that it fixes the issue?

**Comment by graysky2** [comment]
I can confirm that it fixes this issue on my RPi system.  Note that my system has additional packages on it so I feel that someone needs to test this PR on a fresh install in order to verify that none of the other packages I have on my system are needed.  I do not have a spare SD card to do this myself and am hoping someone else in the community will contribute.

**Comment by kmihelich** [comment]
Why are other packages needed to run this package now?

**Comment by graysky2** [comment]
See the `ExecStart` line in the proposed service file.  Those executables are now needed.

**Comment by kmihelich** [comment]
What changed that makes this package only usable from a service file?

**Comment by graysky2** [comment]
1) If you are supplying kodi.service, it should function to shutdown or reboot the machine from the native kodi interface as it has for years.  Recently, this functionality was broken.

2) It is not only usable from a service file.  You may still run the executable from any user with access on the box.

The point of this PR is to fix the bug of the current package NOT being able to shutdown or reboot the box from the Kodi GUI.

**Comment by kmihelich** [comment]
So, then why does this package need to explicitly depend on, and therefore install a swath of extra packages, which are not depended on by any other kodi package?

Additionally, your response shall include a complete justification on *why* this package must now be dependent on xf86-video-fbdev.

**Comment by graysky2** [comment]
1) If you are unwilling to add additional dependencies to this package please consider following Arch upstream and do not provide kodi.service at all.  They have broken it out several months ago.

2) In this case, xorg needs some video driver installed for kodistandalone to run.  For RPi since it does not have a generic video driver, xf86-video-fbdev serves that purpose.  If you try to start the service without that, it will not start and is therefore required.

**Comment by graysky2** [comment]
To my #1 point above, see: https://projects.archlinux.org/svntogit/community.git/commit/trunk?h=packages/xbmc&id=9763c6d32678f3a3f45c195bfae92eee209d504f

**Comment by graysky2** [comment]
Do you have a better solution to https://github.com/archlinuxarm/PKGBUILDs/issues/1140
?

**Comment by kmihelich** [comment]
xf86-video-fbdev is not the only available driver, and I'm very curious why this needs to become the first package to explicitly depend on it.

As for your upstream comparisons, please review exactly how many and what type of packages explicitly depend on xorg-server and xorg-xinit.  Why is it that there are hundreds of packages that run in X that do not depend on either, and what makes this package special?

In addition, you should see the link you provided, as I'm not sure you have.  It deals with a similar, though technically different issue.  It also does nothing to prove why your given X dependencies are required.

This discussion is not about whether I have a solution, it is under your pull request and deals specifically with you providing a complete, non-evasive explanation and justification for the changes you wish to commit to this repository.

**Comment by graysky2** [comment]
Well, this solution worked for me, but maggie2, in the forums has commented that it did NOT work for her.  I just purchased a new SD card and will try this on a fresh install on my RPi to verify or not.

The underlying bug is still present in [alarm]/kodi-rpi ... users starting it with your provided service prevents shutdown or reboots natively from the Kodi GUI.  I am very interested in fixing this bug.  If my PR does not do it, do you have any thoughts?

**Comment by graysky2** [comment]
Nevermind, I have found a solution to this and will make a new PR
